### PR TITLE
Initial check in for dashboard with sections

### PR DESCRIPTION
**Pull request by caradong**
#### Pull request checklist

- initial check in for example dashboard with sections

#### Description of changes

- initial check in for example dashboard with sections

#### Focus areas to test

dashboard example page

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/5899)

Non-Collapsable example:
noncollapseexample

Collapsable example: 
image

image

### CODE REVIEW

**Comment by caradong** [comment]
> 

The work of section component is not finished. Samiya made some progress and I will update this component shortly. 

---
Refers to: packages/dashboard/src/components/Section/Section.tsx:94 in f95f304. [](commit_id = f95f304a80390120e5211b257b5620dd2db955ce, deletion_comment = False)

**Comment by caradong** [comment]
@kkjeer thank you for the instrucutor. Added. in the step to select change, looks like there is only minor and patch. Not sure if I selected the correct one. Is there documentation about providing proper information in thie change file ?

**Comment by kkjeer** [comment]
@caradong for this change, I'd suggest "minor" since this is an addition of new code. "patch" is generally used for bug fixes. Also, the comment in the "description" field goes into release notes. Documentation on writing change files is in our wiki: [https://github.com/OfficeDev/office-ui-fabric-react/wiki/Change-Files](https://github.com/OfficeDev/office-ui-fabric-react/wiki/Change-Files)

---
In reply to: [412674519](https://github.com/OfficeDev/office-ui-fabric-react/pull/5899#issuecomment-412674519) [](ancestors = 412674519)

**Comment by caradong** [comment]
@kkjeer makes sense. I updated the change file as you suggested. Thank you.

File: packages/dashboard/src/components/Section/SectionPage.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          <div>
+            <ExampleCard title="Section" isScrollable={true} isOptIn={true} code={SectionBasicExampleCode}>
+              <SectionBasicExample />
+            </ExampleCard>
+          </div>
+        }
+        propertiesTables={
+          <PropertiesTableSet
+            sources={[require<string>('!raw-loader!@uifabric/dashboard/src/components/Section/Section.types.ts')]}
+            // renderOnly={['IDashboardGridLayoutProps', 'DashboardGridBreakpointLayouts', 'IDashboardCardLayout']}
```
Comment (kkjeer) [inline packages/dashboard/src/components/Section/SectionPage.tsx]: Remove commented line?

File: packages/dashboard/src/components/Section/Section.css [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.react-grid-item {
+  box-shadow: none;
+}
```
Comment (kkjeer) [inline packages/dashboard/src/components/Section/Section.css]: Can this go in Section.styles instead?

Comment (caradong) [reply packages/dashboard/src/components/Section/Section.css]: fixed.

File: packages/dashboard/src/components/Section/SectionPage.tsx (hunk 1 above)
Comment (caradong) [reply packages/dashboard/src/components/Section/SectionPage.tsx]: Done.

---
In reply to: [210771917](https://github.com/OfficeDev/office-ui-fabric-react/pull/5899#discussion_r210771917) [](ancestors = 210771917)

File: packages/dashboard/src/components/Section/Section.css (hunk 2 above)
Comment (caradong) [reply packages/dashboard/src/components/Section/Section.css]: Fixed.

---
In reply to: [210772001](https://github.com/OfficeDev/office-ui-fabric-react/pull/5899#discussion_r210772001) [](ancestors = 210772001)

**Comment by caradong** [comment]
@kkjeer I have resolved all your comments. Please take another look when you get a chance.

File: packages/dashboard/config/pre-copy.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "copyTo": {
-    "lib/components/DashboardGridLayout/": ["./src/components/DashboardGridLayout/DashboardGridLayout.css"]
+    "lib/components/DashboardGridLayout/": ["./src/components/DashboardGridLayout/DashboardGridLayout.css"],
+    "lib/components/Section/": ["./src/components/Section/Section.css"]
```
Comment (kkjeer) [inline packages/dashboard/config/pre-copy.json]: Remove?

File: packages/dashboard/src/components/DashboardGridLayout/examples/DashboardGridLayout.Example.tsx [hunk 4]
```diff
@@ -20,12 +21,19 @@ export class DashboardGridLayoutExample extends React.Component<{}, {}> {
     );
   }
 
+  private _onLayoutChange = (currentLayout: Layout[], allLayouts: Layouts): void => {
```
Comment (kkjeer) [inline packages/dashboard/src/components/DashboardGridLayout/examples/DashboardGridLayout.Example.tsx]: Did you intend to keep these console.log statements?

File: packages/dashboard/src/components/Card/Card.types.ts [hunk 5]
```diff
@@ -11,22 +11,27 @@ export enum CardSize {
   /**
    * Option for selecting small card
    */
-  small,
+  small = 'small',
```
Comment (Raghurk) [inline packages/dashboard/src/components/Card/Card.types.ts]: >small = 'small', [](start = 2, length = 16)

why change this is to string? Have you seen if this breaks the cards in fabric example?

File: packages/dashboard/src/components/DashboardGridLayout/examples/DashboardGridLayout.Example.tsx (hunk 4 above)
Comment (caradong) [reply packages/dashboard/src/components/DashboardGridLayout/examples/DashboardGridLayout.Example.tsx]: I removed the ones that are not necessary but still left a few in the example code. It helps my partners to understand and consume the control. I will remove them in the future if we no longer find them necessary. 

---
In reply to: [211692210](https://github.com/OfficeDev/office-ui-fabric-react/pull/5899#discussion_r211692210) [](ancestors = 211692210)

File: packages/dashboard/src/components/Card/Card.types.ts (hunk 5 above)
Comment (caradong) [reply packages/dashboard/src/components/Card/Card.types.ts]: a few things to clarify:
- this is a string enum in typescript which is quite useful  [https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-4.html](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-4.html)
- this is an effort to merge Size and CardSize and the size enum in DashboardGridLayout.types.ts is already a string enum
- the cards are working fine as far as I can tell.

---
In reply to: [211799641](https://github.com/OfficeDev/office-ui-fabric-react/pull/5899#discussion_r211799641) [](ancestors = 211799641)

**Pull request merged**